examples/chat/client/tasks.nim

```diff
@@ -633,3 +633,99 @@ proc publishWakuChat2*(message: string) {.task(kind=no_rts, stoppable=false).} =
           lightpushHandler)
       else:
         asyncSpawn wakuNode.publish(DefaultTopic, wakuMessage, conf.rlnRelay)
+
+proc getCustomTokens*() {.task(kind=no_rts, stoppable=false).} =
+  let tokens = status.getCustomTokens()
+  if tokens.isErr:
+    let
+      event = GetCustomTokensEvent(error: tokens.error)
+      eventEnc = event.encode
+      task = taskArg.taskName
+
+    trace "task sent errored event to host", event=eventEnc, task
+    asyncSpawn chanSendToHost.send(eventEnc.safe)
+    return
+
+  let
+    event = GetCustomTokensEvent(tokens: tokens.get,
+      timestamp: getTime().toUnix)
+    eventEnc = event.encode
+    task = taskArg.taskName
+
+  trace "task sent event to host", event=eventEnc, task
+  asyncSpawn chanSendToHost.send(eventEnc.safe)
+
+proc addCustomToken*(address: string, name: string, symbol: string, color: string, decimals: string) {.task(kind=no_rts, stoppable=false).} =
+  let timestamp = getTime().toUnix
+
+  if statusState != StatusState.loggedin:
+    let
+      eventNotLoggedIn = AddCustomTokenEvent(error: "Not logged in, " &
+        "cannot add a new custom token.",
+        timestamp: timestamp)
+      eventNotLoggedInEnc = eventNotLoggedIn.encode
+      task = taskArg.taskName
+
+    trace "task sent event to host", event=eventNotLoggedInEnc, task
+    asyncSpawn chanSendToHost.send(eventNotLoggedInEnc.safe)
+    return
+
+  let
+    addResult = status.addCustomToken(address, name, symbol, color, decimals) 
+
+  if addResult.isErr:
+    let
+      event = AddCustomTokenEvent(error: "Error adding a custom token, " &
+        "error: " & addResult.error, timestamp: timestamp)
+      eventEnc = event.encode
+      task = taskArg.taskName
+    trace "task sent event with error to host", event=eventEnc, task
+    asyncSpawn chanSendToHost.send(eventEnc.safe)
+    return
+
+  let
+    token = addResult.get
+    event = AddCustomTokenEvent(address: $token.address,
+      name: token.name, symbol: token.symbol, color: token.color, 
+      decimals: token.decimals, timestamp: timestamp)
+    eventEnc = event.encode
+    task = taskArg.taskName
+
+  trace "task sent event to host", event=eventEnc, task
+  asyncSpawn chanSendToHost.send(eventEnc.safe)
```

Comment:
See comments below, but I think we should be converting types here as satisfying the type needs of the nim-status lib:
```suggestion
  import ../../nim-status/conversions
  
  try: # needed for parseAddress and parseInt, and can also be used to catch any errors with addCustomToken
    let addResult = status.addCustomToken(address.parseAddress, name, symbol, color, decimals.parseInt)
    
    # do the rest...
    
    except Exception as e:
      let
        eventError = AddCustomTokenEvent(error: e.msg, # prob don't wanna add more text, else when the error bubbles, the text will compound
          timestamp: timestamp)
        eventErrorEnc = eventNotLoggedIn.encode
        task = taskArg.taskName

      trace "task sent error to host", event=eventErrorEnc, task
      asyncSpawn chanSendToHost.send(eventErrorEnc.safe)
      return
  
```

Reply:
Thanks, modified as suggested.